cloud: correct interface description according to doc validation (#9631)
pingcap/docs: tidb-cloud/troubleshoot-import-access-denied-error.md

## Check the policy of the IAM role

1. In the AWS Management Console, go to **IAM** > **Access Management** > **Roles**. 
2. In the list of roles, find and click the role you have created for the target TiDB cluster. The role summary page is displayed.

> 3. In the **Permission policies** area of the role summary page, click on each policy to enter the policy summary page.

Author: ```suggestion
3. In the **Permission policies** area of the role summary page, a list of policies is displayed. Click each policy to enter the policy summary page.
```

> 3. In the **Permission policies** area of the role summary page, click on each policy to enter the policy summary page.
> 4. On the policy summary page, click the **{}JSON** tab to see the permission policy. Make sure that the `Resource` fields in each policy are correctly configured.

Author: ```suggestion
3. In the **Permission policies** area of the role summary page, a list of policies is displayed. Take the following steps for each policy:
    1. Click each policy to enter the policy summary page.
    2. On the policy summary page, click the **{}JSON** tab to check the permission policy. Make sure that the `Resource` fields in each policy are correctly configured.
```

## Check the policy of the IAM role

1. In the AWS Management Console, go to **IAM** > **Access Management** > **Roles**, find and click on the role you have created for the target TiDB cluster. The role summary page is displayed.

> 2. In the **Permission policies** area of the role summary page, click on each policy to enter the policy summary page.

Author: @qiancai see https://github.com/pingcap/docs/pull/9459#discussion_r914652554

> **Tip:**
>
> If you have updated the permission policy multiple times and still get the `AccessDenied` error during data import, you can try to revoke active sessions. Go to **IAM** > **Access Management** > **Roles**, click on your target role to enter the role summary page. On the role summary page, find **Revoke active sessions** and click the button to revoke active sessions. Then, retry the data import.
>
> Note that this might affect your other applications.

## Check the bucket policy

1. In the AWS Management Console, open the Amazon S3 console, go to the **Buckets** page. A list of buckets are displayed.
2. Select and click on the target bucket and the bucket information page is displayed.
3. Click the **Permissions** tab and find the **Bucket policy** area. By default, this area has no policy value. If any denied policy is displayed in this area, the `AccessDenied` error might occur during data import.

If you see a denied policy, check whether the policy relates to the current data import. If yes, delete it from the area and retry the data import.

## Check the trust entity

> In the AWS Management Console, go to **IAM** > **Access Management** > **Roles**, find and click on the role you have created for the target TiDB cluster. The role summary page is displayed. On the role summary page, click the **Trust relationships** tab, and you will see the trusted entities.

Reviewer: ```suggestion
1. In the AWS Management Console, go to **IAM** > **Access Management** > **Roles**. 
2. In the list of roles, find and click the role you have created for the target TiDB cluster. The role summary page is displayed. 
3. On the role summary page, click the **Trust relationships** tab, and you will see the trusted entities.
```

## Check the Object Ownership

> 1. In the AWS Management Console, open the Amazon S3 console, go to the **Buckets** page. A list of buckets are displayed.

Reviewer: ```suggestion
1. In the AWS Management Console, open the Amazon S3 console, and then go to the **Buckets** page. A list of buckets is displayed.
```

> 2. Select and click on the target bucket and the bucket information page is displayed.

Reviewer: ```suggestion
2. In the list of buckets, find and click the target bucket. The bucket information page is displayed.
```

> 3. On the bucket information page, scroll down to find the **Object Ownership** area. Make sure that the "Object Ownership" configuration is "Bucket owner enforced".

Reviewer: ```suggestion
3. On the bucket information page, click the **Permissions** tab, and then scroll down to the **Object Ownership** area. Make sure that the "Object Ownership" configuration is "Bucket owner enforced".
```

    If the configuration is not "Bucket owner enforced", the `AccessDenied` error occurs, because your account does not have enough permissions for all objects in this bucket.

To handle the error, click **Edit** in the upper-right corner of the Object Ownership area and change the ownership to "Bucket owner enforced". Note that this might affect your other applications that are using this bucket.

## Check your bucket encryption type

There are more than one way to encrypt an S3 bucket. When you try to access the objects in a bucket, the role you have created must have the permission to access the encryption key for data decryption. Otherwise, the `AccessDenied` error occurs.

> To check the encryption type of your bucket, open the Amazon S3 console in the AWS Management console and click **Buckets** on the left navigation bar, choose the name of the target bucket, choose **Properties**, and you will see the **Default encryption** area that shows the encryption type of the bucket.

Reviewer: ```suggestion
To check the encryption type of your bucket, take the following steps:

1. In the AWS Management Console, open the Amazon S3 console, and then go to the **Buckets** page. A list of buckets is displayed.
2. In the list of buckets, find and click the target bucket. The bucket information page is displayed.
3. On the bucket information page, click the **Properties** tab, scroll down to the **Default encryption** area, and then check the configurations in this area.
```

There are two types of server-side encryption: Amazon S3-managed key (SSE-S3) and AWS Key Management Service (SSE-KMS). For SSE-S3, further check is not needed because this encryption type does not cause access denied errors. For SSE-KMS, you need to check the following:

- If the AWS KMS key ARN in the area is displayed in black without an underline, the AWS KMS key is an AWS-managed key (aws/s3).

> - If the AWS KMS key ARN in the area is displayed in blue with an underline, click on the key ARN to open the key information page. Check the left navigation bar to see the specific encryption type. It might be an AWS managed key (aws/s3) or a customer-managed key.

Reviewer: ```suggestion
- If the AWS KMS key ARN in the area is displayed in blue with a link, click the key ARN to open the key information page. Check the left navigation bar to see the specific encryption type. It might be an AWS managed key (aws/s3) or a customer managed key.
```

<details>
<summary>For the AWS managed key (aws/s3) in SSE-KMS</summary>

In this situation, if the `AccessDenied` error occurs, the reason might be that the key is read-only and cross-account permission grants are not allowed. See the AWS article [Why are cross-account users getting Access Denied errors when they try to access S3 objects encrypted by a custom AWS KMS key](https://aws.amazon.com/premiumsupport/knowledge-center/cross-account-access-denied-error-s3/) for details.

> To solve the access denied error, click **Edit** in the upper-right corner of the **Default encryption** area, and change the AWS KMS key to "Choose from your AWS KMS keys" or "Enter AWS KMS key ARN", or change the server-side encryption method to "AWS S3 Managed Key (SSE-S3). In addition to this method, you can also create a new bucket and use the custom-managed key or the SSE-S3 encryption method.

Reviewer: ```suggestion
To solve the access denied error, click **Edit** in the upper-right corner of the **Default encryption** area, and change the AWS KMS key to "Choose from your AWS KMS keys" or "Enter AWS KMS key ARN", or change the server-side encryption type to "AWS S3 Managed Key (SSE-S3). In addition to this method, you can also create a new bucket and use the custom-managed key or the SSE-S3 encryption method.
```
